saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentServiceIntTest.java

```diff
@@ -72,6 +83,40 @@ public void findOneAndEnrich() {
         assertEquals("123 tys zł 27 gr", enrichedJudgment.getMaxMoneyAmount().getText());
     }
     
+    @Test
+    public void unenrichAndSave() {
+        // given
+        Judgment enrichedJudgment = judgmentEnrichmentService.findOneAndEnrich(judgment.getId());
+
+
+        // execute
+        judgmentEnrichmentService.unenrichAndSave(enrichedJudgment);
+        
+
+        // assert
+        Judgment unenrichedJudgment = judgmentRepository.findOneAndInitialize(judgment.getId());
+        assertEquals(0, judgment.getReferencedCourtCases().size());
+        assertEquals(0, unenrichedJudgment.getReferencedCourtCases().size());
+
+        assertNull(unenrichedJudgment.getMaxMoneyAmount());
+
+        assertEquals(3, judgment.getReferencedRegulations().size());
+        assertEquals(3, unenrichedJudgment.getReferencedRegulations().size());
+        assertEquals(0, unenrichedJudgment.getReferencedRegulations().stream().filter(rr->rr.isGenerated()).count());
+
+    }
+    
+    
+    @Test(expected = GeneratedEntityMergeException.class)
+    public void findOneAndEnrich_GENERATED_SAVE_EXCEPTION() {
+        
+        // given
+        Judgment enrichedJudgment = judgmentEnrichmentService.findOneAndEnrich(judgment.getId());
+        
+        // execute
+        judgmentRepository.save(enrichedJudgment);
```

Comment:
I wonder what can be the result of saving not judgment but one of its generated referenced regulations.